### PR TITLE
Add support for ECS serialisation names.

### PR DESCRIPTION
**Pull request by RubyNova**

### CODE REVIEW

**Comment by RubyNova** [comment]
PLEASE NOTE: This is doing some crazy variadic template stuff. We should hold off on merging this until CI gets updated and fixed. I want to make sure Ubuntu isn't gonna get angry with me.

**Comment by RubyNova** [comment]
This is now also blocked by #430

File: include/NovelRT/Ecs/Configurator.h [hunk 1]
```diff
@@ -28,22 +28,28 @@ namespace NovelRT::Ecs
 
         inline void AddDefaultComponentsAndSystems(SystemScheduler& target)
         {
-            target.GetComponentCache().RegisterComponentType(Graphics::RenderComponent{0, 0, 0, 0, true});
+            target.GetComponentCache().RegisterComponentType(Graphics::RenderComponent{0, 0, 0, 0, true},
+                                                             "NovelRT::Ecs::Graphics::RenderComponent");
```
Comment (capnkenny) [inline include/NovelRT/Ecs/Configurator.h:32]: Is this naming convention a requirement? Or can the user use any unique string?

Comment (RubyNova) [reply include/NovelRT/Ecs/Configurator.h:32]: It can use any unique string - namespaces just made the most sense to my brain.

**Review by capnkenny — APPROVED** [review]
LGTM

**Pull request merged**